app/models/embed/hierarchical_contents.rb

```diff
@@ -22,7 +22,7 @@ def contents
     attr_reader :resources
 
     def file_contents
-      resources.map(&:files).flatten
+      resources.map(&:files).flatten.map { |file| HierarchicalFile.new(file) }
```

Comment:
Could we consider `flat_map` here?

Reply:
You could for `flat_map(&:files)`, but that isn't really the part I'm changing here.